aspnetcore/grpc/versioning.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Versioning gRPC services
+author: jamesnk
+description: Learn how to version gRPC services.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: jamesnk
+ms.date: 01/09/2020
+uid: grpc/versioning
+---
+# Versioning gRPC services
+
+By [James Newton-King](https://twitter.com/jamesnk)
+
+New features added to an app can require gRPC services provided to clients to change, sometimes in unexpected and breaking ways. When gRPC services change:
+
+* Consideration should be given on how changes impact clients.
+* A versioning strategy to support changes should be implemented.
+
+## Backwards compatibility
+
+The gRPC protocol is designed to support services that change over time. Generally additions to gRPC services and methods are non-breaking. Non-breaking means existing clients continue to work. Changing or deleting gRPC services are breaking changes. Breaking changes mean existing clients fail.
+
+Making non-breaking changes to a service has a number of benefits:
+
+- Existing clients continue to run.
+- Avoids work involved with notifying clients of breaking changes, and updating them.
+- Only one version of the service needs to be documented and maintained.
+
+This content focuses on whether changes are **breaking at a gRPC protocol and .NET binary compatibility level**. When making changes, consider whether older clients can logically continue working. For example, adding a new field to a request message:
+
+* Is not a protocol breaking change.
+* Setting an error condition if the new field is not set makes it a breaking change for old clients.
+
+### Non-breaking changes
+
+These changes are non-breaking at a gRPC protocol level, and .NET binary level.
+
+- **Adding a new service**
+- **Adding a new method to a service**
+- **Adding a field to a request message** - Fields added to a request message are deserialized with the [default value](https://developers.google.com/protocol-buffers/docs/proto3#default) on the server when not set.
+- **Adding a field to a response message** - Fields added to a response message are deserialized into the message's [unknown fields](https://developers.google.com/protocol-buffers/docs/proto3#unknowns) collection on the client.
+- **Adding a value to an enum** - Enums are serialized as a numeric value. New enum values are deserialized on the client to the enum value without an enum name.
```

Comment:
I feel like this needs a little bit more elucidation - it's correct that this is not a protocol level break - but it probably represents a behavior change that the client is not prepared for.